app/models/comment.rb

```diff
@@ -0,0 +1,17 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post
+
+  validates :text, presence: true
+
+  before_validation :set_timestamps
+
+  def set_timestamps
+    self.created_at ||= Time.current
+    self.updated_at = Time.current
+  end
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ Required ] I think it would be nice if you ensure that all custom methods in your models are also tested. To comply with the project requirements please add q unit test for the `update_comments_counter` method in the `Comment` model.

## Example code suggestion for update_comments_counter_spec
```ruby
require 'rails_helper'

RSpec.describe Comment, type: :model do
  before(:all) do
    @user = User.create(
      name: 'Anne',
      photo: 'https://anne.com/ann.jpg',
      bio: 'This Annes bio',
      posts_counter: 0
    )
    @post = Post.create(
      title: 'A new dawn',
      text: 'Software Developer',
      author_id: @user.id,
      comments_counter: 0,
      likes_counter: 0
    )
  end

  it 'Users comments_counter should increase by one' do
    Comment.create(text: 'Hello there!!', author_id: @user.id, post_id: @post.id)
    Comment.create(text: 'Welcome home!!', author_id: @user.id, post_id: @post.id)
    post = Post.find(@post.id)
    expect(post.update_comments_counter).to eq(2)
  end
end
```

## Project requirements
![Screenshot from 2023-11-18 13-10-59](https://github.com/paulshaun-kdtk/blog-posts/assets/16392046/3884496b-33ca-40bd-b132-dbf684fa13c0)